plugins/ipam/host-local/backend/disk/backend.go

```diff
@@ -108,13 +110,36 @@ func (s *Store) ReleaseByID(id string) error {
 		if err != nil {
 			return nil
 		}
-		if strings.TrimSpace(string(data)) == strings.TrimSpace(id) {
+		if strings.TrimSpace(string(data)) == (strings.TrimSpace(id) + "\n" + ifname) {
 			if err := os.Remove(path); err != nil {
 				return nil
 			}
+			found = true
 		}
 		return nil
 	})
+
+	if (!found) && (err == nil) {
```

Comment:
You actually don't need the () around !found since it's a bool already.

Also, maybe a comment here explaining that this block is for backwards compatibility when reading files written by a previous version of the plugin that didn't store interface name?

Reply:
Please ensure you've run `go fmt`.  It should remove those unnecessary parens (and help prevent other stylistic inconsistencies).